includes/blocks/course-theme/class-lesson-actions.php

```diff
@@ -0,0 +1,153 @@
+<?php
+/**
+ * File containing the Lesson_Actions class.
+ *
+ * @package sensei
+ * @since
+ */
+
+namespace Sensei\Blocks\Course_Theme;
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit;
+}
+
+use \Sensei_Blocks;
+use \Sensei_Course;
+use \Sensei_Lesson;
+use \Sensei_Utils;
+
+/**
+ * Class Lesson_Actions is responsible for rendering the Lesson actions block.
+ */
+class Lesson_Actions {
+	/**
+	 * Lesson_Actions constructor.
+	 */
+	public function __construct() {
+		Sensei_Blocks::register_sensei_block(
+			'sensei-lms/course-theme-lesson-actions',
+			[
+				'render_callback' => [ $this, 'render' ],
+			]
+		);
+	}
+
+	/**
+	 * Renders complete lesson button.
+	 *
+	 * @param int  $lesson_id                   Lesson ID.
+	 * @param bool $has_incomplete_prerequisite Whether it has incomplete prerequisite.
+	 *
+	 * @return string The complete lesson button.
+	 */
+	private function render_complete_lesson( $lesson_id, $has_incomplete_prerequisite ) {
+		if (
+			// Return empty if the lesson requires a quiz pass.
+			Sensei()->lesson->lesson_has_quiz_with_questions_and_pass_required( $lesson_id )
+		) {
+			return '';
+		}
+
+		// The button is disabled if the lesson requires a pre-requisite.
+		$disabled = $has_incomplete_prerequisite ? 'disabled' : '';
+
+		// The button is a secondary CTA if there is a quiz but not required to take/pass it.
+		$button_style = 'is-primary';
+		if ( Sensei_Lesson::lesson_quiz_has_questions( $lesson_id ) ) {
+			$button_style = 'is-secondary';
```

Comment:
The idea with using one block is to move all this logic, like checking if there is another button, out of the individual button rendering functions, so it's checked once and the parent block decides which buttons to show and what is primary/secondary/disabled, and just passes that along to the button rendering functions. 
Otherwise we could end up with inconsistent states if some check is only changed in one of the buttons.

Reply:
I did a refactor here moving all the checks to the main method: https://github.com/Automattic/sensei/pull/4490/commits/486e7ea7faf195208b0a55e38b047c4518d22201

Reply:
Nice!